### PR TITLE
NEW: Stack component

### PR DESCRIPTION
**Pull request by wEprous**
This Pull Request meets the following criteria:

* [x] Tests have been added/adjusted for my new feature
* [x] New Components are registered in index.js of my project

### CODE REVIEW

File: src/ButtonGroup/index.js [hunk 1]
```diff
@@ -10,6 +10,7 @@ import type { Props } from "./index";
 
 const StyledButtonGroup = styled.div`
   display: flex;
+  width: 100%;
```
Comment (tomashapl) [inline src/ButtonGroup/index.js]: Why this ?

File: src/FormFeedback/index.js [hunk 2]
```diff
@@ -14,8 +14,11 @@ const StyledFormFeedback = styled(({ theme, type, ...props }) => <div {...props}
   font-size: ${({ theme }) => theme.orbit.fontSizeFormFeedback};
   font-weight: ${({ theme, type }) =>
     type === TYPE_OPTIONS.ERROR ? theme.orbit.fontWeightMedium : theme.orbit.fontWeightNormal};
-  margin-top: ${({ theme }) => theme.orbit.marginTopFormFeedback};
   line-height: ${({ theme }) => theme.orbit.lineHeightText};
+  width: 100%;
+  margin-top: 2px;
+  position: absolute;
+  bottom: -${({ theme }) => Math.floor(theme.orbit.lineHeightText * parseInt(theme.orbit.fontSizeFormFeedback, 10)) + 2}px;
```
Comment (tomashapl) [inline src/FormFeedback/index.js]: Nice 👏

File: src/List/ListItem/index.js [hunk 3]
```diff
@@ -16,6 +16,9 @@ export const Item = styled.div`
   &:last-child {
```
Comment (tomashapl) [inline src/List/ListItem/index.js]: please merge it to one nested selector: 

```
&:last-child, &:last-of-type
```

File: src/Radio/index.js [hunk 4]
```diff
@@ -27,7 +28,9 @@ Glyph.defaultProps = {
 const IconContainer = styled.div`
   position: relative;
   box-sizing: border-box;
+  flex: 0 0 auto;
   display: flex;
+  flex-shrink: 0;
```
Comment (tomashapl) [inline src/Radio/index.js]: duplication of setting `flex-shrink`

File: src/Stack/index.js [hunk 5]
```diff
@@ -0,0 +1,201 @@
...
+import media from "../utils/media";
+import defaultTokens from "../defaultTokens";
+import { ALIGNS, DIRECTIONS, SPACINGS, TOKENS } from "./consts";
+import { StyledButton } from "../Button";
+import getSpacingToken from "../common/getSpacingToken";
+
+import type { Props } from "./index";
+
+// TODO: use tokens for spacings
+const getSpacing = (spacing, direction = DIRECTIONS.COLUMN, type) => () => {
```
Comment (tomashapl) [inline src/Stack/index.js]: please receive object:

```
{ spacing, type, direction }
```

File: src/Stack/index.js [hunk 6]
```diff
@@ -0,0 +1,201 @@
...
+    flex && (direction === DIRECTIONS.ROW ? "row" : "column")};
+  flex-wrap: ${({ wrap, flex }) => flex && (wrap ? "wrap" : "nowrap")};
+  flex-grow: ${({ grow, flex }) => flex && (grow ? "1" : "0")};
+  flex-shrink: ${({ shrink, flex }) => flex && (shrink ? "1" : "0")};
+  flex-basis: ${({ basis, flex }) => flex && basis};
+  justify-content: ${({ flex, align, direction }) =>
+    flex && (direction === DIRECTIONS.ROW || align) && getAlign(align || "start")};
+  align-items: ${({ flex }) => flex && "stretch"};
+  align-content: ${({ align, direction, flex }) =>
+    flex && (direction === DIRECTIONS.COLUMN || align) && getAlign(align || "start")};
```
Comment (tomashapl) [inline src/Stack/index.js]: Can we move fallback directly into function ?

File: src/Stack/index.js [hunk 7]
```diff
@@ -0,0 +1,201 @@
...
+    margin: ${({ spacing, direction, align }) =>
+      align !== ALIGNS.EVEN && getSpacing(spacing, direction, TOKENS.MOBILESPACING)};
+    &:last-child {
+      margin: 0;
+    }
+  }
+
+  // TODO other block components
+  & > ${StyledButton} {
+    align-self: ${({ align }) => getAlign(align === ALIGNS.EVEN ? ALIGNS.START : align)};
```
Comment (tomashapl) [inline src/Stack/index.js]: This logic could by in function

File: src/Stack/index.js [hunk 8]
```diff
@@ -0,0 +1,201 @@
...
+            desktop.basis};
+          justify-content: ${({ flex, align, direction }) =>
+            flex &&
+            desktop &&
+            desktop.direction &&
+            (direction !== desktop.direction || align !== desktop.align) &&
+            desktop.direction === DIRECTIONS.ROW &&
+            getAlign(desktop.align || align)};
+          align-content: ${({ align, direction, flex }) =>
+            flex &&
```
Comment (tomashapl) [inline src/Stack/index.js]: Uff 😓there are repeating conditions, we should create const for this

**Comment by tomashapl** [comment]
Should we add some tests like

```
.toHaveStyleRule("vertical-align", "middle");
``` 

?

File: src/common/getSpacingToken/index.js [hunk 9]
```diff
@@ -0,0 +1,18 @@
+// @flow
+import SPACINGS_AFTER from "./consts";
+
+import type { Props } from ".";
+
+const getSpacingToken = () => ({ spacingAfter, theme }: Props) => {
```
[the remark this answers is not in the document]
Comment (wEprous) [reply src/common/getSpacingToken/index.js]: OK, did you test it outside storybook? Just that we are 100 percent safe with usage of it :)

[the remark this answers is not in the document]
Comment (tomashapl) [reply src/common/getSpacingToken/index.js]: I don't see reason why storybook should change behavior of JS funcs 😀 you can check `getRenderSize` func in `StyledCarrierLogo` there is applied this case

File: src/ButtonGroup/index.js (hunk 1 above)
Comment (wEprous) [reply src/ButtonGroup/index.js]: Good catch :)

File: src/Stack/index.js (hunk 8 above)
Comment (tomashapl) [reply src/Stack/index.js]: 👍

**Pull request merged**